packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2072,6 +2074,22 @@ class RichText extends LeafRenderObjectWidget {
   }
 }
 
+/// A TextSpan that opens [url] in the default browser. If [text] is null the url is displayed.
+///
+/// See also:
+///
+/// [TextSpan]
+/// [UrlLancher.launch]
+class LinkTextSpan extends TextSpan {
+  LinkTextSpan({ TextStyle style, @required String url, String text }) : super(
+    style: style,
+    text: text ?? url,
+    recognizer: new TapGestureRecognizer()..onTap = () {
+      UrlLauncher.launch(url);
+    }
+  );
```

Comment:
The tricky thing here is that you need to dispose this recognizer when you remove the LinkTextSpan from the tree.